taqueria-vscode-extension-web-ui/src/data-editors/ListEditor.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { DataEditorNode } from './DataEditorNode';
+import { compare, validate } from './MichelineValidator';
+import { ValidationResultDisplay } from './ValidationResultDisplay';
+import { VSCodeButton } from './VsCodeWebViewUIToolkitWrappers';
+
+export const ListEditor = (
+	{ dataType, value, onChange }: { dataType: any; value: any[]; onChange: (value: any) => void },
+) => {
+	if (value === undefined || value === null || !Array.isArray(value)) {
+		value = [];
+		onChange(value);
+	}
+	const changeValue = (index: number, v: any) => {
+		const newValue = value.slice();
+		newValue[index] = v;
+		onChange(newValue);
+	};
+	const moveUp = (index: number) => {
+		if (index <= 0) {
+			return;
+		}
+		const newValue = value.slice();
+		newValue[index - 1] = value[index];
+		newValue[index] = value[index - 1];
+		onChange(newValue);
+	};
+	const moveDown = (index: number) => {
+		if (index >= value.length - 1) {
+			return;
+		}
+		const newValue = value.slice();
+		newValue[index + 1] = value[index];
+		newValue[index] = value[index + 1];
+		onChange(newValue);
+	};
+	const remove = (index: number) => {
+		const newValue = value.slice();
+		newValue.splice(index, 1);
+		onChange(newValue);
+	};
+	const sort = () => {
```

Comment:
Is this meant to sort in reverse order? If so, naming it `sortInDescending` makes more sense.

e.g. `const sortInDescending = () => ...`

Reply:
No. It actually sorts ascending. Just that my compare has a different convention to the JS's compare. (when which argument is greater it returns +1). Maybe it's better to fix that instead

Reply:
Fixed